### PR TITLE
Conform to recipe standards

### PR DESCRIPTION
**Pull request by boldorider4**
### Description

Change build script in recipe generation to conform to latest recipe curation standards.

### CODE REVIEW

**Comment by jezdez** [comment]
@boldorider4 Thanks for the patch, please sign the CLA as indicated above by the conda bot. Thank you!

File: conda_build/skeletons/pypi.py [hunk 1]
```diff
@@ -431,7 +431,7 @@ def skeletonize(
             if noarch_python:
                 ordered_recipe["build"]["noarch"] = "python"
 
-            recipe_script_cmd = ["{{ PYTHON }} -m pip install . -vv"]
+            recipe_script_cmd = ["{{ PYTHON }} -m pip install . --vv --no-deps --no-build-isolation"]
```
Comment (kenodegard) [inline conda_build/skeletons/pypi.py:434]:
```suggestion
            recipe_script_cmd = ["{{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation"]
```

File: news/4960-conform-to-recipe-standards [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+* <news item>
+
+### Bug fixes
+
+* <news item>
+
+### Deprecations
+
+* when building new recipes from a pypi packag, the build script `{{ PYTHON }} -m pip install . --vv` is deprecated in favor of `{{ PYTHON }} -m pip install . --vv --no-deps --no-build-isolation`
```
Comment (kenodegard) [inline news/4960-conform-to-recipe-standards:11]:
```suggestion
* When templating new recipes from a pypi package, the build script `{{ PYTHON }} -m pip install . -vv` is deprecated in favor of `{{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation`. (#4960)
```

**Comment by kenodegard** [comment]
pre-commit.ci autofix

**Comment by kenodegard** [comment]
@conda-bot check

**Pull request merged**